### DISCUSSION TITLE
I'm trying to add Traits to a component using .addTrait function but the thing is on reloading the window the Traits get disappeared?

### DISCUSSION THREAD

**Discussion by daud99** [GrapesJS/grapesjs#3903] [Q&A]
```
let gridComponent = component.append(requiredGrid)[0];
gridComponent.addTrait([columnsTrait, rowsTrait, columnGapTrait, rowGapTrait, minTrait]);
```

So, these two lines are included in the function which gets execute to add a new grid which is getting done successfully and I can also see the Traits as expected. However, when I reload the window of browser the Traits are gone. However, the grid is their but with only default Traits i.e. Id and title. I want to persist them how can I do it.

**Comment by artf** [comment]
Traits are not storable, they only follow the Component Definition (this allows you to have upgradable Traits).
If you need different Traits for some reason, you have to implement the add/update/delete logic inside the component itself (on init, on change of some prop, etc.).
